Test/UpdateUserView.swift

```diff
@@ -9,8 +9,12 @@ import ComposableArchitecture
 import SwiftUI
 
 struct UpdateUserState: Equatable {
+  init(user: User?) {
+    self.user = user
+    self.username = user?.username ?? ""
```

Comment:
This is the only part that I dont fully like/get it. Its better to do this rather that sending the action `.onAppear` and updating the `state.username` in the reducer?

Reply:
You have to remember to it in `onAppear`. If your state gets an user, I don't think there's a reason for the username to be different until `.onAppear`, isn't it?

Reply:
yeah thats a good reason